client/config/config.go

```diff
@@ -92,220 +48,104 @@ func Cmd(defaultCLIHome string) *cobra.Command {
 
 	cmd.Flags().String(flags.FlagHome, defaultCLIHome,
 		"set client's home directory for configuration")
-	cmd.Flags().Bool(flagGet, false,
-		"print configuration value or its default if unset")
+	//	cmd.Flags().Bool(flagGet, false,
+	//		"print configuration value or its default if unset")
 	return cmd
 }
 
 func runConfigCmd(cmd *cobra.Command, args []string) error {
 
-	
-	InitConfigTemplate()
+	v := viper.New()
 
-	cfgPath, err := ensureConfFile(viper.GetString(flags.FlagHome))
+	cfgPath, err := ensureCfgPath(v.GetString(flags.FlagHome))
 	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to make config path: %v\n", err)
 		return err
 	}
 
-	getAction := viper.GetBool(flagGet)
-	if getAction && len(args) != 1 {
-		return ErrWrongNumberOfArgs
-	}
-
-	cliConfig, err := getClientConfig(cfgPath)
+	cliConfig, err := getClientConfig(cfgPath, v)
 	if err != nil {
-		return fmt.Errorf("Unable to get client config %v", err)
+		fmt.Fprintf(os.Stderr, "Unable to get client config: %v\n", err)
+		return err
 	}
 
-
 	switch len(args) {
 	case 0:
 		// print all client config fields to sdt out
-		// TODO implement method to print out all client config fiels
+		// TODO implement method to print out all client config fields
 		fmt.Println(cliConfig.ChainID)
 		fmt.Println(cliConfig.KeyringBackend)
 		fmt.Println(cliConfig.Output)
 		fmt.Println(cliConfig.Node)
 		fmt.Println(cliConfig.BroadcastMode)
 		fmt.Println(cliConfig.Trace)
-		
+
 	case 1:
-		// it's a get  
+		// it's a get
 		// TODO implement method for get
+		// should i implement getters here?
 		key := args[0]
 		switch key {
-			case "chain-id": fmt.Println(cliConfig.ChainID)
-			case "keyring-backend": fmt.Println(cliConfig.KeyringBackend)
-			case "output": fmt.Println(cliConfig.Output)
-			case "node": fmt.Println(cliConfig.Node)
-			case "broadcast-mode": fmt.Println(cliConfig.BroadcastMode)
-			case "trace": fmt.Println(cliConfig.Trace)
-			default :
-				return errUnknownConfigKey(key)
+		case flags.FlagChainID:
+			fmt.Println(cliConfig.ChainID)
+		case flags.FlagKeyringBackend:
+			fmt.Println(cliConfig.KeyringBackend)
+		case tmcli.OutputFlag:
+			fmt.Println(cliConfig.Output)
+		case flags.FlagNode:
+			fmt.Println(cliConfig.Node)
+		case flags.FlagBroadcastMode:
+			fmt.Println(cliConfig.BroadcastMode)
+		case "trace":
```

Comment:
Could not find hardcoded flag for "trace" ((

Reply:
I found `server.FlagTrace`

Reply:
Thanks